zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -70,6 +70,23 @@
     private final PreparedStatement selectTraceDurationTail;
     private final Map<String,String> metadata;
 
+    private final ThreadLocal<Set<String>> writtenNames = new ThreadLocal<Set<String>>() {
+                private long hour = TimeUnit.MILLISECONDS.toHours(System.currentTimeMillis());
```

Comment:
the size of the set is a concern.
i can't imagine the number of service and span names within any one hour goes to such an unweldy number though.

if it is a problem a simple way to solve it is just limit the size of the set and let writes spill over.

a guava cache introduces concurrency which as i tried to explain above is just overhead.

> Another possible scenario would be getting cache misses if the data is in the Set in another thread.

not a problem really. a cache miss only means that the write happens. today we have a 100% cache miss and it's "not a problem" :-)